File: bindings/java/src/main/java/com/satox/sdk/CoreManager.java
package com.satox.sdk;

/**
 * Core Manager for the Satox SDK
 * 
 * Manages the core functionality of the SDK including initialization,
 * configuration, and status monitoring.
 */
public class CoreManager {
    private final long handle;
    
    CoreManager(long handle) {
        this.handle = handle;
    }
    
    /**
     * Initialize the core manager
     * 
     * @throws SatoxException if initialization fails
     */
    public void initialize() throws SatoxException {
        int result = SatoxSDK.initializeCoreManager(handle);
        if (result != 0) {
            throw new SatoxException("Failed to initialize core manager");
        }
    }
    
    /**
     * Shutdown the core manager
     */
    public void shutdown() {
        SatoxSDK.shutdownCoreManager(handle);
    }
    
    /**
     * Get the status of the core manager
     * 
     * @return Status JSON string
     */
    public String getStatus() {
        return SatoxSDK.getCoreManagerStatus(handle);
    }
    
    /**
     * Get the configuration of the core manager
     * 
     * @return Configuration JSON string
     */
    public String getConfig() {
        return SatoxSDK.getCoreManagerConfig(handle);
    }
    
    /**
     * Set the configuration of the core manager
     * 
     * @param config Configuration JSON string
     * @throws SatoxException if setting config fails
     */
    public void setConfig(String config) throws SatoxException {
        int result = SatoxSDK.setCoreManagerConfig(handle, config);
        if (result != 0) {
            throw new SatoxException("Failed to set core manager config");
        }
    }
    
    /**
     * Get the native handle
     * 
     * @return Native handle
     */
    long getHandle() {
        return handle;
    }
} 

File: bindings/typescript/src/managers/core-manager.ts
/**
 * @file core-manager.ts
 * @brief Core Manager for TypeScript/JavaScript bindings
 * @copyright Copyright (c) 2025 Satoxcoin Core Developers
 * @license MIT License
 */

import { EventEmitter } from 'events';
import { native } from '../../build/Release/satox_sdk_native';
import {
  CoreConfig,
  IManager,
  SDKEvent,
  SatoxError,
  SatoxCoreError,
  AsyncBoolean,
  AsyncString,
  AsyncNumber,
  AsyncVoid
} from '../types';

/**
 * Core Manager providing core SDK functionality
 */
export class CoreManager extends EventEmitter implements IManager {
  private _initialized = false;
  private _config: CoreConfig = {};
  private _lastError = '';
  private _stats: Record<string, any> = {};

  constructor() {
    super();
    this.resetStats();
  }

  /**
   * Initialize the core manager
   */
  public async initialize(config: CoreConfig): Promise<boolean> {
    try {
      if (this._initialized) {
        throw new SatoxCoreError('Core manager already initialized');
      }

      this._config = { ...config };
      
      // Initialize native core manager
      const result = native.coreManagerInitialize(JSON.stringify(config));
      if (!result.success) {
        throw new SatoxCoreError(result.error || 'Failed to initialize native core manager');
      }

      this._initialized = true;
      this.emit('initialized', { timestamp: new Date().toISOString() });

      return true;
    } catch (error) {
      this._lastError = error.message;
      const satoxError = error instanceof SatoxError ? error : new SatoxCoreError(error.message);
      this.emit('error', satoxError);
      throw satoxError;
    }
  }

  /**
   * Shutdown the core manager
   */
  public async shutdown(): Promise<void> {
    try {
      if (!this._initialized) {
        return;
      }

      native.coreManagerShutdown();
      this._initialized = false;
      this.emit('shutdown', { timestamp: new Date().toISOString() });
    } catch (error) {
      this._lastError = error.message;
      const satoxError = error instanceof SatoxError ? error : new SatoxCoreError(error.message);
      this.emit('error', satoxError);
      throw satoxError;
    }
  }

  /**
   * Check if core manager is initialized
   */
  public isInitialized(): boolean {
    return this._initialized;
  }

  /**
   * Get last error
   */
  public getLastError(): string {
    return this._lastError;
  }

  /**
   * Get statistics
   */
  public getStats(): Record<string, any> {
    return { ...this._stats };
  }

  /**
   * Reset statistics
   */
  public resetStats(): void {
    this._stats = {
      operations_performed: 0,
      errors_encountered: 0,
      last_operation: null,
      uptime: 0
    };
  }

  /**
   * Health check for core manager
   */
  public async healthCheck(): Promise<boolean> {
    try {
      if (!this._initialized) {
        return false;
      }

      const result = native.coreManagerHealthCheck();
      return result.success;
    } catch {
      return false;
    }
  }

  /**
   * Get SDK version
   */
  public async getVersion(): Promise<string> {
    try {
      if (!this._initialized) {
        throw new SatoxCoreError('Core manager not initialized');
      }

      const result = native.coreManagerGetVersion();
      if (!result.success) {
        throw new SatoxCoreError(result.error || 'Failed to get version');
      }

      return result.version;
    } catch (error) {
      this._lastError = error.message;
      const satoxError = error instanceof SatoxError ? error : new SatoxCoreError(error.message);
      this.emit('error', satoxError);
      throw satoxError;
    }
  }

  /**
   * Get configuration
   */
  public getConfig(): CoreConfig {
    return { ...this._config };
  }

  /**
   * Update configuration
   */
  public async updateConfig(config: Partial<CoreConfig>): Promise<boolean> {
    try {
      if (!this._initialized) {
        throw new SatoxCoreError('Core manager not initialized');
      }

      const newConfig = { ...this._config, ...config };
      const result = native.coreManagerUpdateConfig(JSON.stringify(newConfig));
      if (!result.success) {
        throw new SatoxCoreError(result.error || 'Failed to update configuration');
      }

      this._config = newConfig;
      this._stats.operations_performed++;
      this._stats.last_operation = 'update_config';

      this.emit('config_updated', {
        timestamp: new Date().toISOString(),
        config: newConfig
      });

      return true;
    } catch (error) {
      this._lastError = error.message;
      this._stats.errors_encountered++;
      const satoxError = error instanceof SatoxError ? error : new SatoxCoreError(error.message);
      this.emit('error', satoxError);
      throw satoxError;
    }
  }
} 